web/src/activity.ts

```diff
@@ -191,16 +193,54 @@ export function mark_client_active(): void {
     }
 }
 
+export function setup_browser_idle_detector(idle_handler: {cancel: () => void}): void {
+    // This code is separated out of `initialize` to facilitate testing
+    if (!browser_idle_detection.is_browser_idle_detector_supported()) {
+        blueslip.log("Browser idle detector not supported");
+        return;
+    }
+
+    $(document).one("keypress click", () => {
+        void browser_idle_detection.request_browser_idle_detector_permission();
+    });
+
+    browser_idle_detection.listen_to_idle_detector_permission_change(() => {
+        void browser_idle_detection
+            .init_browser_idle_detector({
+                idle_timeout: 60 * 1000,
+                on_idle: mark_client_idle,
+                on_active: mark_client_active,
+            })
+            .then((result) => {
+                if (result === "started") {
+                    // We no longer need to track idle state with jQuery.
+                    idle_handler.cancel();
+                    blueslip.info("Browser IdleDetector started");
+                    return;
+                }
+                if (result.name !== "NotAllowedError") {
+                    // We don't need to handle `NotAllowedError` specifically
+                    // since we have requested for the permission above, and
+                    // when we do get permission, the change handler
+                    // will call this callback again.
+                    blueslip.error("Browser IdleDetector failed to start: " + result.message);
+                }
+            });
+    });
+}
+
 export function initialize(): void {
     $("html").on("mousemove", () => {
         set_new_user_input(true);
     });
 
     $(window).on("focus", mark_client_active);
-    $(window).idle({
+    const idle_handler = $(window).idle({
         idle: DEFAULT_IDLE_TIMEOUT_MS,
         onIdle: mark_client_idle,
         onActive: mark_client_active,
         keepTracking: true,
     });
+
+    setup_browser_idle_detector(idle_handler);
```

Comment:
I'd recommend renaming this method to `setup_idle_handler()` and then move the existing JQuery `$().idle()` call inside our function as well.

Essentially, that function would handle all of the idling behavior.